### PR TITLE
157246332 Add error message to user signup page

### PR DESCRIPTION
**Pull request by Spectrumsun**
#### What does this PR do?
add error handling to user signup page
add image preview to the signup page

#### Description of Task to be completed?
a user should get an error message if the username or email is already used
a preview of the image is display on the page

#### How should this be manually tested?
on the signup page, the user will get an error message if the email or password is already used

#### Any background context you want to provide?
User doesn't get any notification or error if the username or email is already used. make it a confusing for the user

#### What are the relevant pivotal tracker stories?
#157283077

#### Screenshots (if appropriate)
non

#### Questions:
non

### CODE REVIEW

File: public/js/controllers/index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ angular.module('mean.system')
     };
 
     $scope.login = () => {
-      console.log("SCOPE USER",$scope.user);
+      //console.log("SCOPE USER",$scope.user);
```
Comment (amarachukwu-agbo) [inline public/js/controllers/index.js:20]: Good job Sunday on adding the error message for user signup page. I can see a commented out console statement and I suggest you remove it from your code as the team agreed.

File: public/js/controllers/index.js [hunk 2]
```diff
@@ -40,51 +40,69 @@ angular.module('mean.system')
...
-              }
-            );
-        })
-        .catch((err) => {
-          alert('Unable to upload. Check your internet', err);
-        });
-    };
+      $http.post('/api/auth/checkuser', $scope.user)
+      .then((res) => { 
+        //console.log(res.status)
```
Comment (amarachukwu-agbo) [inline public/js/controllers/index.js:45]: The same feedback applies here as well.

File: public/js/controllers/index.js [hunk 3]
```diff
@@ -40,51 +40,69 @@ angular.module('mean.system')
...
+        $scope.error = error.data.error
+
+      })
+
+    };    
+
+
+
+    $scope.imageUpload = function (event) {
+      var files = event.target.files;
```
Comment (kenoseni) [inline public/js/controllers/index.js:90]: Good job Sunday, you should use es6 syntax of declaring variables with const or let.

File: public/js/controllers/index.js [hunk 4]
```diff
@@ -40,51 +40,69 @@ angular.module('mean.system')
...
+      for (var i = 0; i < files.length; i++) {
+          var file = files[i];
```
Comment (kenoseni) [inline public/js/controllers/index.js:92]: Same apply here, please avoid the use of var. Use const or let instead

File: public/js/controllers/index.js [hunk 5]
```diff
@@ -40,51 +40,69 @@ angular.module('mean.system')
...
+          var reader = new FileReader();
```
Comment (kenoseni) [inline public/js/controllers/index.js:93]: Please view comment above and implement accordingly

**Review by kenoseni — COMMENTED** [review]
Good job fixing the alert pop-up that we initially see when the username and email already exist in the database, now we get an error message on the page. I have left a few comments below, please check them and  make changes.

**Pull request closed without merging**